AwsCryptographicMaterialProviders/runtimes/rust/tests/local_cmc_tests.rs

```diff
@@ -0,0 +1,147 @@
+    use aws_mpl_rs::operation::get_cache_entry::GetCacheEntryOutput;
+    use aws_mpl_rs::types::error::Error;
+    use aws_mpl_rs::types::Materials;
+    use aws_mpl_rs::deps::aws_cryptography_keyStore::types::BeaconKeyMaterials;
+    use std::collections::HashMap;
+    use chrono::Utc;
+    use rand::{thread_rng, Rng};
+    use futures::stream::StreamExt;
+
+    async fn work_for_thread() -> Result<(), crate::BoxError> {
+        let mpl_config: MaterialProvidersConfig = MaterialProvidersConfig::builder().build()?;
+        let mpl: mpl_client::Client = mpl_client::Client::from_conf(mpl_config)?;
+
+        let test: CryptographicMaterialsCacheRef = mpl.create_cryptographic_materials_cache()
+            .cache(CacheType::Default(
+                DefaultCache::builder()
+                .entry_capacity(10)
+                .build()?))
+            .send()
+            .await?;
+
+        let identifiers: Vec<&str> = vec![
+            "one",
+            "two",
+            "three",
+            "four",
+            "five",
+            "six",
+            "seven",
+            "eight",
+            "nine",
+            "ten",
+            "eleven",
+            "twelve",
+            "thirteen",
+            "fourteen",
+            "fifteen",
+            "sixteen",
+            "seventeen",
+            "eighteen",
+            "nineteen",
+            "twenty",
+            "twenty one"
+        ];
+
+        let id_size: usize = identifiers.len();
+
+        let invocations = 300000;
+        let concurrent_tasks = 10;
+        
+        futures::stream::iter(0..invocations)
+            .map(|_i| {
+                let identifiers = identifiers.clone();
+                let test = test.clone();
+
+                async move {
+                    println!("TestLotsofAdd");
+                    let random = thread_rng().gen_range(0..id_size);
+                    let beacon_key_identifier: &str = identifiers[random];
+
+                    let cache_identifier: aws_smithy_types::Blob = aws_smithy_types::Blob::new(beacon_key_identifier.as_bytes());
+
+                    let cache_entry_output: Result<GetCacheEntryOutput, Error> = test.get_cache_entry()
+                        .identifier(cache_identifier.clone())
+                        .send()
+                        .await;
+
+                    match cache_entry_output {
+                        Ok(_) => {
+                            println!("Cache hit for beacon key identifier: {}", beacon_key_identifier);
+                        }
+                        Err(Error::EntryDoesNotExist {message: _m}) => {
+                            let materials = Materials::BeaconKey(
+                                    BeaconKeyMaterials::builder()
+                                    .beacon_key_identifier(beacon_key_identifier.to_string())
+                                    .beacon_key(cache_identifier.clone())
```

Comment:
Add the comments from .NET and Java

Reply:
I still have to push a couple of changes. Will add this with that.